go/src/koding/kites/kloud/provider/do/machine_test.go

```diff
@@ -0,0 +1,244 @@
+		m, ok := machine.(*Machine)
+		if !ok {
+			return fmt.Errorf("can't type assert %T to *Machine", machine)
+		}
+
+		m.client = client
+
+		ctx := context.Background()
+		_, err := m.Stop(ctx)
+		return err
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestMachineInfo(t *testing.T) {
+	tests := []struct {
+		name       string
+		dropletID  int // by default initialized
+		statusCode int // by default 200
+		status     string
+		state      machinestate.State
+	}{
+		{
+			name:   "new droplet",
+			status: "new",
+			state:  machinestate.Starting,
+		},
+		{
+			name:   "running droplet",
+			status: "active",
+			state:  machinestate.Running,
+		},
+		{
+			name:   "stopped droplet",
+			status: "off",
+			state:  machinestate.Stopped,
+		},
+		{
+			name:   "destroyed droplet",
+			status: "archive",
+			state:  machinestate.Terminated,
+		},
+		{
+			name:       "not available droplet",
+			state:      machinestate.NotInitialized,
+			statusCode: http.StatusNotFound,
+		},
+		{
+			name:       "server not responding",
+			state:      machinestate.Unknown,
+			statusCode: http.StatusInternalServerError,
+		},
+		{
+			name:      "droplet id malformed",
+			dropletID: -1,
+			state:     machinestate.Unknown,
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			dropletID := 12345
+			if test.dropletID != 0 {
+				dropletID = 0
+			}
+
+			bm := newDoBaseMachine(dropletID)
+			machine, err := newMachine(bm)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			handler := func(w http.ResponseWriter, r *http.Request) {
+				droplet := &godo.Droplet{
+					ID:     dropletID,
+					Status: test.status,
+				}
+
+				if test.statusCode != 0 {
+					w.WriteHeader(test.statusCode)
+				}
+				_ = json.NewEncoder(w).Encode(newRootDroplet(droplet))
+
+			}
+
+			err = withClient(http.HandlerFunc(handler), func(client *godo.Client) error {
+				m, ok := machine.(*Machine)
+				if !ok {
+					return fmt.Errorf("can't type assert %T to *Machine", machine)
+				}
+
+				m.client = client
+
+				ctx := context.Background()
+				state, _, _ := m.Info(ctx)
+				if state != test.state {
+					return fmt.Errorf("expecting %q, got: %q", test.state, state)
+				}
+
+				return nil
+			})
+			if err != nil {
+				t.Fatal(err)
+			}
+		})
+	}
+}
+
+func withClient(handler http.Handler, fn func(client *godo.Client) error) error {
+	server := httptest.NewServer(handler)
+	defer server.Close()
+
+	client := godo.NewClient(nil)
+	url, _ := url.Parse(server.URL)
+	client.BaseURL = url
+
+	return fn(client)
+}
+
+func newDoBaseMachine(dropletID int) *provider.BaseMachine {
+	return &provider.BaseMachine{
+		Machine:    &models.Machine{},
+		Session:    nil,
+		Credential: doProvider.Schema.NewCredential(),
+		Bootstrap:  doProvider.Schema.NewBootstrap(),
+		Metadata: doProvider.Schema.NewMetadata(&stack.Machine{
+			Attributes: map[string]string{
+				"droplet_id": strconv.Itoa(dropletID),
+			},
+		}),
+		Provider: "do",
```

Comment:
i was not sure if this field is used to create terraform template where they use `digitalocean` as a prefix [link](https://www.terraform.io/docs/providers/do/r/droplet.html).